lib/simulator-xcode-9.js

```diff
@@ -97,6 +133,49 @@ class SimulatorXcode9 extends SimulatorXcode8 {
   }
 
   /**
+   * Update the common iOS Simulator preferences file with new values.
+   * It is necessary to restart the corresponding Simulator before
+   * these changes are applied.
+   *
+   * @param {DevicePreferences} devicePrefs [{}] - The mapping, which represents new device preference values
+   *                                               for the given Simulator.
+   * @param {CommonPreferences} commonPrefs [{}] - The mapping, which represents new common preference values
+   *                                               for all Simulators.
+   * @return {boolean} True if the preferences were successfully updated.
+   */
+  async updatePreferences (devicePrefs = {}, commonPrefs = {}) {
+    if (!_.isEmpty(devicePrefs)) {
+      log.debug(`Setting preferences of ${this.udid} Simulator to ${JSON.stringify(devicePrefs)}`);
+    }
+    if (!_.isEmpty(commonPrefs)) {
+      log.debug(`Setting common Simulator preferences to ${JSON.stringify(commonPrefs)}`);
+    }
+    const homeFolderPath = process.env('HOME');
+    if (!homeFolderPath) {
+      log.warn(`Cannot get the path to HOME folder from the process environment. ` +
+        `Ignoring Simulator preferences update.`);
+      return false;
+    }
+    const plistPath = path.resolve(homeFolderPath, 'Library', 'Preferences', 'com.apple.iphonesimulator.plist');
+    if (!await fs.hasAccess(plistPath)) {
+      log.warn(`Simulator preferences file ${plistPath} is not accessible. ` +
```

Comment:
Paths in particular should be in single quotes in logs, since strange characters can get in there and cause problems sometimes, so seeing what the _actual_ value is is important.

Reply:
fixed